web-common/src/features/canvas/components/ComponentRenderer.svelte

```diff
@@ -0,0 +1,39 @@
+<script lang="ts">
+  import { Chart } from "./charts";
+  import { Image } from "./image";
+  import { KPI } from "./kpi";
+  import { Markdown } from "./markdown";
+  import { Table } from "./table";
+
+  import {
+    createQueryServiceResolveComponent,
+    type V1ComponentSpecRendererProperties,
+  } from "@rilldata/web-common/runtime-client";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+
+  export let renderer: string;
+  export let componentName: string;
+
+  $: componentQuery = createQueryServiceResolveComponent(
+    $runtime.instanceId,
+    componentName,
+    { args: {} },
+  );
+  $: componentData = $componentQuery?.data;
+  $: rendererProperties =
+    componentData?.rendererProperties as V1ComponentSpecRendererProperties;
+</script>
+
+{#if rendererProperties}
+  {#if renderer === "kpi"}
```

Comment:
Can we move to a map instead of if-else. Will be cleaner once we have a lot of components. Similar to how we select the workspace here: https://github.com/rilldata/rill/blob/main/web-local/src/routes/(application)/(workspace)/files/%5B...file%5D/%2Bpage.svelte#L56